web-backend/ui/src/main.rs

```diff
@@ -153,6 +151,59 @@ fn valid_repo_url(repo: &str) -> bool {
     true
 }
 
+// Fetches the list of exisiting repositories in analysis database
+async fn get_all_repos() -> Result<Vec<Repo>> {
+    let db = Db::new(None, None, None, None).await.unwrap();
+    let config = Config::new(db);
+    let repos = match config.get_repos().await {
```

Comment:
You should probably pass in a DB here instead of creating a new one

Reply:
In other places, we pass `app.clone()` which contains a DB and `db.clone()`, effectively creating new instances of DB anyways. Is there any difference between passing a DB clone and creating a new DB inside?

Reply:
Its more about ensuring that we're using the same DB that the App is instead of creating a new one thats not associated with the application. behind the scenes it may not be much different but this makes it clear that we're using the DB the app is